react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable no-console */
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: ''
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({
+      email: event.target.value
+    });
+  }
+
+  handleSubmit(event) {
+    console.log(this.state);
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label>
+          Email
+        </label>
+        <input type="email" onChange={this.handleChange}></input>
```

Comment:
Be sure to also pass a `value` prop to your input so that it is fully "controlled".

Reply:
revised!

Reply:
I'm unclear as to exactly what value prop does though.  would you be able to clarify?

Reply:
The idea is that React should know about and dictate the value in the input at any given time. This becomes more relevant in situations where you want to preload some initial form values, like if a user wants to edit something they had previously saved.